pkg/ddevapp/silverstripe.go

```diff
@@ -0,0 +1,67 @@
+package ddevapp
+
+import (
+	"fmt"
+	"github.com/ddev/ddev/pkg/fileutil"
+	"github.com/ddev/ddev/pkg/nodeps"
+	"github.com/ddev/ddev/pkg/util"
+	"os"
+	"path/filepath"
+)
+
+// isSilverstripeApp returns true if the app is of type Silverstripe
+func isSilverstripeApp(app *DdevApp) bool {
+	return fileutil.FileExists(filepath.Join(app.AppRoot, app.ComposerRoot, "vendor/bin/sake"))
+}
+
+func silverstripePostStartAction(app *DdevApp) error {
+	// We won't touch env if disable_settings_management: true
+	if app.DisableSettingsManagement {
+		return nil
+	}
+	envFilePath := filepath.Join(app.AppRoot, app.ComposerRoot, ".env")
+	_, envText, err := ReadProjectEnvFile(envFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("Unable to read .env file: %v", err)
+	}
+	if os.IsNotExist(err) {
+		err = fileutil.CopyFile(filepath.Join(app.AppRoot, app.ComposerRoot, ".env.example"), envFilePath)
+		if err != nil {
+			util.Debug("Silverstripe: .env.example does not exist yet, not trying to process it")
+			return nil
+		}
+		_, envText, err = ReadProjectEnvFile(envFilePath)
+		if err != nil {
+			return err
+		}
+	}
+	port := "3306"
+	dbConnection := "MySQLDatabase"
+	if app.Database.Type == nodeps.Postgres {
+		dbConnection = "PostgreSQLDatabase"
+		port = "5432"
+	}
+	envMap := map[string]string{
```

Comment:
I wonder would it make sense to add values for`SS_DEFAULT_ADMIN_USERNAME` and `SS_DEFAULT_ADMIN_PASSWORD` as well to the to be created `.env` file? I am in no way an Silverstripe expert but it looked the one or two times I've installed it recently that initially there is no user created by Silverstripe out of the box and the quickstart guide also  recommends setting the credentials for the default user: https://docs.silverstripe.org/en/5/getting_started/

Reply:
I did think about that too. Given it's .env, and for local development, it's probably good to do so. 👍

Reply:
cool! if there would be another way or a username and pw preset i haven't found that would also totally fine. but when i've skimmed through the docs to setup silverstripe i couldnt find anything, the getting started and it's instructions for the .env file were the only guidance. 
and from a new users perspective for someone who is unfamiliar with a CMS such a quick start is super convenient to get around potential stepping stones. that is something i really appreciate from any CMS documentation.

p.s. and thanks for working on that PR! was evaluating and comparing many CMSes recently, silverstripe was one of them, and this project type would have saved me some head scratches :D

Reply:
Thanks :)

If you have any questions, feel free to flick me an email via my address here on GitHub :) I do believe I know a thing or two about Silverstripe... Since... I kinda worked with it for 10 years and have worked at the actual company for 4 years too, in which I rewrote the whole Security implementation, among things :)

Reply:
I'm sure impressed! I'm sure you'll be successful.

Reply:
Setting aside potential typos, @rpkoller , I think I've added it properly now. :)

Reply:
oh awesome! i've left two follow-up comments in the context of the quickstarter.md now that the default username and password are set. one is about informing the user what the set username and password actually (which is inline with the example used for Moodle). plus i've suggested to add `ddev launch /admin`at the end. so that the person installing the CMS is directly forwarded to the url where it is possible to log into the admin interface since all other recipes use that.